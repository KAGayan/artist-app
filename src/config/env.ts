export const env = {
  API_KEY: process.env.REACT_APP_API_KEY,
  BASE_URL: process.env.REACT_APP_API_BASE_URL,
  ARTIST: process.env.REACT_APP_ARTIST,
};
