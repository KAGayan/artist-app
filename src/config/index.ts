export * from './env';
export * from './makeClasses';
