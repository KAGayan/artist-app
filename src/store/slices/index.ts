import { topalbums, albumsActions } from './albumsSlice';
import { albumInfo, albumInfoActions } from './albumInfoSlice';

export const reducer = {
  topalbums,
  albumInfo,
};

export const actions = {
  ...albumsActions,
  ...albumInfoActions,
};
