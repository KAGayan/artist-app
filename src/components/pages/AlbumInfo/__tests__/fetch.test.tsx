import { rest } from 'msw';
import { setupServer } from 'msw/node';
import { render, screen, waitFor } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import { ReactElement } from 'react';
import { store } from 'store';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import { env } from 'config';
import { AlbumInfo } from '..';

const server = setupServer(
  rest.get(
    `${env.BASE_URL}/`,
    (req, res, ctx) => res(ctx.json({
      album: {
        artist: 'Eminem', listeners: '119874', image: [{ size: 'small', '#text': '' }, { size: 'medium', '#text': '' }, { size: 'large', '#text': '' }, { size: 'extralarge', '#text': '' }, { size: 'mega', '#text': '' }, { size: '', '#text': '' }], mbid: '', tags: '', name: '(null)', playcount: '1879721', url: 'https:/www.last.fm/music/Eminem/(null)',
      },
    })),
  ),
);

interface Props {
  children: ReactElement;
  reduxStore: any;
}

const ReduxProvider = ({ children, reduxStore }: Props) => (
  <Provider store={reduxStore}>{children}</Provider>
);

const mockedUsedNavigate = jest.fn();
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom') as any,
  useNavigate: () => mockedUsedNavigate,
}));

window.scrollTo = jest.fn();

beforeAll(() => server.listen());
afterEach(() => {
  server.resetHandlers();
  jest.resetAllMocks();
});
afterAll(() => {
  server.close();
  jest.clearAllMocks();
});

test('loads and displays Album Not Found!', async () => {
  const history = createMemoryHistory();
  const pageTree = render(
    <Router
      location={history.location}
      navigator={history}
    >
      <ReduxProvider reduxStore={store}>
        <AlbumInfo />
      </ReduxProvider>
    </Router>,
  );

  await waitFor(() => screen.getByRole('alert'));

  expect(screen.getByRole('alert')).toHaveTextContent('Album Not Found!');
  expect(pageTree.container).toMatchSnapshot();
});
