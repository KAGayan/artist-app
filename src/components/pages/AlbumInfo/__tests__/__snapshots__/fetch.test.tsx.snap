// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loads and displays Album Not Found! 1`] = `
<div>
  <div
    class="css-urx51g-MuiStack-root"
  >
    <header
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-11b3ww9-MuiPaper-root-MuiAppBar-root"
    >
      <div
        class="MuiContainer-root MuiContainer-maxWidthXl css-19r6kue-MuiContainer-root"
      >
        <div
          class="MuiToolbar-root MuiToolbar-regular css-1cndkqf-MuiToolbar-root"
        >
          <div
            class="css-1o5q8ri-MuiStack-root"
          >
            <button
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1d159sf-MuiSvgIcon-root"
                data-testid="ArrowBackIosNewOutlinedIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M17.77 3.77 16 2 6 12l10 10 1.77-1.77L9.54 12z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
          <div
            class="css-eyffu-MuiStack-root"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              Eminem
            </p>
            <div
              class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-1qxe5qy-MuiAvatar-root"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiAvatar-fallback css-10mi8st-MuiSvgIcon-root-MuiAvatar-fallback"
                data-testid="PersonIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </header>
    <div
      class="css-1n75z56-MuiStack-root"
    >
      <div
        class="MuiContainer-root MuiContainer-maxWidthXl css-19r6kue-MuiContainer-root"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-12n0uxm-MuiGrid-root"
        >
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-standardError MuiAlert-standard css-xilmwd-MuiPaper-root-MuiAlert-root"
            role="alert"
          >
            <div
              class="MuiAlert-icon css-1ytlwq5-MuiAlert-icon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                data-testid="ErrorOutlineIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                />
              </svg>
            </div>
            <div
              class="MuiAlert-message css-acap47-MuiAlert-message"
            >
              Album Not Found!
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
